### ISSUE TITLE
Unable to convert multiple files with the same engine with include processor

### ISSUE THREAD

**Issue by SevereOverfl0w** [asciidoctor/asciidoctorj#579]
Asciidoctor Version: 1.6.0-alpha5

```
clojure.lang.ExceptionInfo: asciidoctor: FAILED: /home/dominic/.boot/cache/tmp/home/dominic/src/github.com/SevereOverfl0w/experiments/query-tests/hx2/-jnm801/bidi.adoc: Failed to load AsciiDoc document {:file "/tmp/boot.user6726404817545392910.clj", :line 51}
	at clojure.core$ex_info.invokeStatic(core.clj:4725)
	at clojure.core$ex_info.invoke(core.clj:4725)
	at boot.main$_main$fn__1788.invoke(main.clj:213)
	at boot.main$_main.invoke(main.clj:206)
	at clojure.lang.Var.invoke(Var.java:396)
	at org.projectodd.shimdandy.impl.ClojureRuntimeShimImpl.invoke(ClojureRuntimeShimImpl.java:159)
	at org.projectodd.shimdandy.impl.ClojureRuntimeShimImpl.invoke(ClojureRuntimeShimImpl.java:150)
	at boot.App.runBoot(App.java:399)
	at boot.App.main(App.java:491)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at Boot.main(Boot.java:257)
Caused by: java.lang.IllegalStateException: asciidoctor: FAILED: /home/dominic/.boot/cache/tmp/home/dominic/src/github.com/SevereOverfl0w/experiments/query-tests/hx2/-jnm801/bidi.adoc: Failed to load AsciiDoc document
	at org.asciidoctor.extension.Processor.setConfig(Processor.java:145)
	at org.asciidoctor.extension.processorproxies.IncludeProcessorProxy.initialize(IncludeProcessorProxy.java:80)
	at org.asciidoctor.extension.processorproxies.IncludeProcessorProxy$INVOKER$i$1$0$initialize.call(IncludeProcessorProxy$INVOKER$i$1$0$initialize.gen)
	at org.jruby.internal.runtime.methods.JavaMethod$JavaMethodOneOrN.call(JavaMethod.java:850)
	at org.jruby.runtime.callsite.CachingCallSite.callBlock(CachingCallSite.java:171)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:177)
	at org.jruby.RubyClass.newInstance(RubyClass.java:1001)
	at org.jruby.RubyClass$INVOKER$i$newInstance.call(RubyClass$INVOKER$i$newInstance.gen)
	at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:200)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:161)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:314)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:73)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:109)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:95)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:298)
	at org.jruby.runtime.callsite.CachingCallSite.callBlock(CachingCallSite.java:79)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:83)
	at org.jruby.ir.instructions.CallBase.interpret(CallBase.java:428)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:355)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:73)
	at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:83)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:179)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:165)
	at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:200)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:338)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:163)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:314)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:73)
	at org.jruby.ir.interpreter.Interpreter.INTERPRET_BLOCK(Interpreter.java:132)
	at org.jruby.runtime.MixedModeIRBlockBody.commonYieldPath(MixedModeIRBlockBody.java:148)
	at org.jruby.runtime.IRBlockBody.doYield(IRBlockBody.java:193)
	at org.jruby.runtime.BlockBody.yield(BlockBody.java:125)
	at org.jruby.runtime.Block.yieldArray(Block.java:177)
	at org.jruby.RubyBasicObject.yieldUnder(RubyBasicObject.java:1754)
	at org.jruby.RubyBasicObject.instance_exec19(RubyBasicObject.java:1728)
	at org.jruby.RubyBasicObject$INVOKER$i$0$3$instance_exec19.call(RubyBasicObject$INVOKER$i$0$3$instance_exec19.gen)
	at org.jruby.runtime.callsite.CachingCallSite.callBlock(CachingCallSite.java:77)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:83)
	at org.jruby.ir.instructions.CallBase.interpret(CallBase.java:428)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:355)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:73)
	at org.jruby.ir.interpreter.Interpreter.INTERPRET_BLOCK(Interpreter.java:132)
	at org.jruby.runtime.MixedModeIRBlockBody.commonYieldPath(MixedModeIRBlockBody.java:148)
	at org.jruby.runtime.IRBlockBody.doYield(IRBlockBody.java:186)
	at org.jruby.runtime.BlockBody.yield(BlockBody.java:116)
	at org.jruby.runtime.Block.yield(Block.java:165)
	at org.jruby.RubyArray.each(RubyArray.java:1734)
	at org.jruby.RubyArray$INVOKER$i$0$0$each.call(RubyArray$INVOKER$i$0$0$each.gen)
	at org.jruby.internal.runtime.methods.JavaMethod$JavaMethodZeroBlock.call(JavaMethod.java:498)
	at org.jruby.runtime.callsite.CachingCallSite.callBlock(CachingCallSite.java:77)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:83)
	at org.jruby.ir.instructions.CallBase.interpret(CallBase.java:428)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:355)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:73)
	at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:83)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:179)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:165)
	at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:200)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:161)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:314)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:73)
	at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:89)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:214)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:200)
	at org.jruby.runtime.callsite.CachingCallSite.callBlock(CachingCallSite.java:203)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:209)
	at org.jruby.RubyClass.newInstance(RubyClass.java:1008)
	at org.jruby.RubyClass$INVOKER$i$newInstance.call(RubyClass$INVOKER$i$newInstance.gen)
	at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:208)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:358)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:195)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:323)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:73)
	at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:89)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:214)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:200)
	at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:208)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:193)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:323)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:73)
	at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:89)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:214)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:200)
	at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:208)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:193)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:323)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:73)
	at org.jruby.ir.interpreter.Interpreter.INTERPRET_BLOCK(Interpreter.java:132)
	at org.jruby.runtime.MixedModeIRBlockBody.commonYieldPath(MixedModeIRBlockBody.java:148)
	at org.jruby.runtime.IRBlockBody.doYield(IRBlockBody.java:186)
	at org.jruby.runtime.BlockBody.yield(BlockBody.java:116)
	at org.jruby.runtime.Block.yield(Block.java:165)
	at org.jruby.RubyIO.ensureYieldClose(RubyIO.java:1163)
	at org.jruby.RubyIO.open(RubyIO.java:1156)
	at org.jruby.RubyIO$INVOKER$s$0$0$open.call(RubyIO$INVOKER$s$0$0$open.gen)
	at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:204)
	at org.jruby.runtime.callsite.CachingCallSite.callBlock(CachingCallSite.java:171)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:177)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:332)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:73)
	at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:89)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:214)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:200)
	at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:208)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:193)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:323)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:73)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:109)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:95)
	at org.jruby.gen.InterfaceImpl1180378365.convertFile(org/jruby/gen/InterfaceImpl1180378365.gen:13)
	at org.asciidoctor.internal.JRubyAsciidoctor.convertFile(JRubyAsciidoctor.java:525)
	at org.asciidoctor.internal.JRubyAsciidoctor.convertFile(JRubyAsciidoctor.java:548)
	at org.asciidoctor.internal.JRubyAsciidoctor.convertFile(JRubyAsciidoctor.java:543)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at clojure.lang.Reflector.invokeMatchingMethod(Reflector.java:93)
	at clojure.lang.Reflector.invokeInstanceMethod(Reflector.java:28)
	at myproject.main$render.invokeStatic(main.clj:41)
	at myproject.main$render.invoke(main.clj:39)
	at myproject.main$main$fn__324.invoke(main.clj:49)
	at clojure.core$run_BANG_$fn__8233.invoke(core.clj:7590)
	at clojure.core.protocols$fn__7665.invokeStatic(protocols.clj:167)
	at clojure.core.protocols$fn__7665.invoke(protocols.clj:124)
	at clojure.core.protocols$fn__7620$G__7615__7629.invoke(protocols.clj:19)
	at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:31)
	at clojure.core.protocols$fn__7648.invokeStatic(protocols.clj:75)
	at clojure.core.protocols$fn__7648.invoke(protocols.clj:75)
	at clojure.core.protocols$fn__7594$G__7589__7607.invoke(protocols.clj:13)
	at clojure.core$reduce.invokeStatic(core.clj:6704)
	at clojure.core$run_BANG_.invokeStatic(core.clj:7585)
	at clojure.core$run_BANG_.invoke(core.clj:7585)
	at myproject.main$main.invokeStatic(main.clj:49)
	at myproject.main$main.invoke(main.clj:43)
	at boot.user$eval334$fn__335$fn__343$fn__344.invoke(boot.user6726404817545392910.clj:44)
	at boot.core$run_tasks.invoke(core.clj:1019)
	at boot.core$boot$fn__918.invoke(core.clj:1029)
	at clojure.core$binding_conveyor_fn$fn__5297.invoke(core.clj:2027)
	at clojure.lang.AFn.call(AFn.java:18)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalStateException: It is only allowed to set the config in the constructor!
	... 144 more
```

I am using clojure, so no easy reproduction case is available. The gist is that I create an engine via Asciidoctor.Factory, and register an IncludeProcessor. It's very similar to the uri include processor ruby example.

When doing `.convertFile` on a second file, I get the above exception.

I can try and write a java example, but I suspect that someone who writes java regularly will be able to do this far quicker than I could.

**Comment by robertpanzer** [comment]
Hmm... I thought I had fixed this with the latest version.
I'll check.

**Comment by robertpanzer** [comment]
I can reproduce the problem, seems like I fixed it only for BlockProcessors 😖 
Will create a fix asap.

**Comment by robertpanzer** [comment]
Created PR #580 to fix this problem for the remaining extension types.

**Comment by SevereOverfl0w** [comment]
What would the release schedule be like for this PR?

**Comment by robertpanzer** [comment]
We're still alpha and there hasn't been much happening since the last release. 
Imo it should be fine to make a new release soon.
I am currently on a trip though and not able to do a release (with signing the jars) before I am back home.
Therefore I could do it in about a week.

**Comment by robertpanzer** [comment]
Once the PR is merged, a SNAPSHOT release should be published automatically to jcenter though.
That would allow you to test it earlier.

**Comment by SevereOverfl0w** [comment]
That would be great. Thanks for your time on this.

**Comment by robertpanzer** [comment]
Should be fixed now with 1.6.0-alpha.6.
